rust/agama-server/src/manager/web.rs

```diff
@@ -226,36 +228,53 @@ async fn installer_status(
     Ok(Json(status))
 }
 
-/// Returns agama logs
-#[utoipa::path(get, path = "/api/manager/logs", responses(
-  (status = 200, description = "Download logs blob.")
+/// Creates router for handling /logs/* endpoints
+fn logs_router() -> Router<ManagerState<'static>> {
+    Router::new()
+        .route("/store", get(download_logs))
+        .route("/list", get(list_logs))
+}
+
+#[utoipa::path(get, path = "/logs/store", responses(
+    (status = 200, description = "Compressed Agama logs", content_type="application/octet-stream"),
 ))]
 
-pub async fn download_logs() -> impl IntoResponse {
-    let path = generate_logs().await;
-    let Ok(path) = path else {
-        return (StatusCode::INTERNAL_SERVER_ERROR).into_response();
-    };
+async fn download_logs() -> impl IntoResponse {
+    let mut headers = HeaderMap::new();
 
-    match ServeFile::new(path)
-        .try_call(Request::new(axum::body::Body::empty()))
-        .await
-    {
-        Ok(res) => res.into_response(),
-        Err(_) => (StatusCode::INTERNAL_SERVER_ERROR).into_response(),
-    }
-}
+    match storeLogs(LogOptions::default()) {
```

Comment:
Always passing `LogOptions::default` makes me think whether the arguments are useful in this case.